apps/platform/src/components/dashboard/secret/secretRevisionSheet/index.tsx

```diff
@@ -110,6 +112,32 @@ export default function SecretRevisionsSheet(): React.JSX.Element {
     }
   }, [environments, getAllRevisionsOfSecret, selectedSecret, setRevisions])
 
+  useEffect(() => {
+    if (!projectPrivateKey || !revisions.length) return
+
+    revisions.forEach(({ versions }) => {
+      versions.forEach((revision) => {
+        if (decryptedRevisions[revision.version]) return
+
+        decrypt(projectPrivateKey, revision.value)
```

Comment:
if projectPrivateKey is present, call the decrypt functon, and add the value into the final array. If not, add 'Hidden' and move on